public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -894,15 +1066,23 @@ export class LokiDatasource
     return returnVariables(expr);
   }
 
-  // Used when running queries through backend
+  /**
+   * Filters out queries that are empty or hidden. Used when running queries through backend.
+   * It is called from DatasourceWithBackend.
+   * @returns `true` if the query is not hidden and its expression is not empty; `false` otherwise.
+   */
   filterQuery(query: LokiQuery): boolean {
```

Comment:
Unrelated, but this is a very unfortunate name for a data source method with a very specific use case for `DatasourceWithBackend`, where as "filter query" sounds very generic.